### PR TITLE
New: Headless Spawn module as an alternative to headless_ai if any pr…

### PR DESCRIPTION
**Pull request by EM-Creations**
**When merged this pull request will:**
- Headless Spawn module as an alternative to headless_ai if any problems are encountered.

### CODE REVIEW

File: modules/headless_spawn/init.sqf [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ * headless AI units will spawn in and what behaviour they'll have
+ */
+#include "script_component.hpp"
+
+#include "settings.sqf"
+
+[{CBA_missionTime > 0}, {
+  params["_groups"];
+
+  private _hcPresent = if (isNil "HC_1") then {false} else {true};
```
Comment (PiZZAD0X) [inline modules/headless_spawn/init.sqf:12]:
```suggestion
  private _hcPresent = isNil "HC_1";
```

File: modules/headless_spawn/init.sqf [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  doSpawn = {
```
Comment (PiZZAD0X) [inline modules/headless_spawn/init.sqf:14]:
```suggestion
  _doSpawn = {
```

File: modules/headless_spawn/init.sqf [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+      /* }; */
+      _run = _run + 1;
+    };
+  };
+
+  // Leave this code alone
+  if (_hcPresent && isMultiplayer) then { // If HC is connected
+    if (!isServer && !hasInterface) then { // If this unit is the HC
+      call doSpawn;
```
Comment (PiZZAD0X) [inline modules/headless_spawn/init.sqf:81]:
```suggestion
      call _doSpawn;
```

File: modules/headless_spawn/init.sqf [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    } else {
+      // Do nothing.
+    };
+  } else {
+    if (isServer) then { // If HC isn't connected and this is the server
+      call doSpawn;
```
Comment (PiZZAD0X) [inline modules/headless_spawn/init.sqf:87]:
```suggestion
      call _doSpawn;
```

File: modules/headless_spawn/init.sqf [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        _crewGroup addWaypoint [(getMarkerPos "marker_wp4"), -1]; */
+
+      /* }; */
+      _run = _run + 1;
+    };
+  };
+
+  // Leave this code alone
+  if (_hcPresent && isMultiplayer) then { // If HC is connected
+    if (!isServer && !hasInterface) then { // If this unit is the HC
```
Comment (PiZZAD0X) [inline modules/headless_spawn/init.sqf:80]:
```suggestion
    if (!isServer && {!hasInterface}) then { // If this unit is the HC
```

File: modules/modules.sqf [hunk 6]
```diff
@@ -46,6 +46,7 @@ Additional modules that can be enabled by removing the //
 //#include "gas\root.sqf"
 //#include "grad-fortifications\root.sqf"
 #include "headless_ai\root.sqf"
```
Comment (PiZZAD0X) [inline modules/modules.sqf:48]:
```suggestion
//#include "headless_ai\root.sqf"
```

Comment (EM-Creations) [reply modules/modules.sqf:48]: You reckon headless AI should no longer be a default modul?

Comment (PiZZAD0X) [reply modules/modules.sqf:48]: yeah, loads things when not at all needed in TVT

File: modules/headless_spawn/init.sqf (hunk 1 above)
Comment (EM-Creations) [reply modules/headless_spawn/init.sqf:12]: Should be !isNil but yes.

**Review by PiZZAD0X — APPROVED** [review]

**Pull request merged**